ml_tools/feature_generator.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+from abc import ABC, abstractmethod
+
+
+class FeatureGenerator(ABC):
+
+    # def __init__(self, data):
+    #     self._data = data.copy()
+
+    @abstractmethod
+    def generate_feature(self, data, column=None, **kwargs):
+        raise NotImplementedError("generate_feature must be implemented")
+
+
+# class GenericGenerator(FeatureGenerator):
+#     def generate_feature(self, data, column=None):
+#         pass
+
+
+class Hour(FeatureGenerator):
+
+    @classmethod
+    def generate_feature(cls, data, column=None, **kwargs):
+        if column is None:
+            types = data.dtypes
+            ts_indices = [idx for idx, t in enumerate(list(types)) if t == "datetime64[ns]"]
```

Comment:
- [ ] docstrings!  We use Google style.
- [ ] Docstrings on public functions with args and returns and raises.   Docstrings on private functions that just have a lite description.